kolibri/core/webpack/hooks.py

```diff
@@ -103,40 +101,43 @@ def _stats_file_content(self):
           written by Webpack.
         """
         cache_key = "json_stats_file_cache_{unique_id}".format(unique_id=self.unique_id)
-        try:
-            stats_file_content = caches[CACHE_NAMESPACE].get(cache_key)
-            if not stats_file_content or getattr(settings, "DEVELOPER_MODE", False):
-                with io.open(self._stats_file, mode="r", encoding="utf-8") as f:
-                    stats = json.load(f)
-                if getattr(settings, "DEVELOPER_MODE", False):
-                    timeout = 0
-                    while stats["status"] == "compiling":
-                        time.sleep(0.1)
-                        timeout += 0.1
-                        with io.open(self._stats_file, mode="r", encoding="utf-8") as f:
-                            stats = json.load(f)
-                        if timeout >= 5:
-                            raise WebpackError("Webpack compilation still in progress")
-                    if stats["status"] == "error":
-                        raise WebpackError("Webpack compilation has errored")
-                stats_file_content = {
-                    "files": stats.get("chunks", {}).get(self.unique_id, []),
-                    "hasMessages": stats.get("messages", False),
-                }
-                # Don't invalidate during runtime.
-                # Might need to change this if we move to a different cache backend.
-                caches[CACHE_NAMESPACE].set(cache_key, stats_file_content, None)
-            return stats_file_content
-        except IOError as e:
-            if hasattr(e, "filename"):
-                problem = "Problems loading: {file}".format(file=e.filename)
-            else:
-                problem = "Not file-related."
-            raise WebpackError(
-                "Webpack build file missing, front-end assets cannot be loaded. {problem}".format(
-                    problem=problem
-                )
-            )
+        stats_file_content = caches[CACHE_NAMESPACE].get(cache_key)
+
+        if not stats_file_content or getattr(settings, "DEVELOPER_MODE", False):
+
+            if not os.path.exists(self._stats_file):
+                raise WebpackError("Missing stats file: '{}'".format(self._stats_file))
+
+            with io.open(self._stats_file, mode="r", encoding="utf-8") as f:
```

Comment:
This seems like it will generate an unhandled `IOError` if the stats file doesn't exist? Can we reinstate the try...except just around this block?

Reply:
Nvm, just saw the above.

Reply:
Although trying and catching is more Pythonic :)

Reply:
yeah, will change to wrap individual lines

The main thing I was doing here was removing the try...catch around the _entire_ block, because it seemed to be re-catching `WebpackError` thrown inside which was causing super confusing behaviors

Reply:
Yup, that was definitely insufficiently granular.